persist/fs/files.go

```diff
@@ -211,29 +328,44 @@ func forEachInfoFile(filePathPrefix string, namespace ident.ID, shard uint32, re
 		if err != nil {
 			continue
 		}
-		fn(matched[i], infoData)
+		fn(matched[i].Files[0], infoData)
```

Comment:
do you not need a len check on Files here?

Reply:
yeah that would be safer, fixed